homeassistant/components/tibber/sensor.py

```diff
@@ -349,6 +383,41 @@ async def _fetch_data(self):
         ]["estimatedAnnualConsumption"]
 
 
+class TibberDataSensor(TibberSensor, update_coordinator.CoordinatorEntity):
+    """Representation of a Tibber sensor."""
+
+    def __init__(
+        self,
+        tibber_home,
+        coordinator: update_coordinator.DataUpdateCoordinator,
+        entity_description: SensorEntityDescription,
+    ):
+        """Initialize the sensor."""
+        super().__init__(coordinator=coordinator, tibber_home=tibber_home)
+        self.entity_description = entity_description
+
+        self._attr_unique_id = (
+            f"{self._tibber_home.home_id}_{self.entity_description.key}"
+        )
+        self._attr_name = f"{entity_description.name} {self._home_name}"
+        if entity_description.key == "month_cost":
+            self._attr_native_unit_of_measurement = self._tibber_home.currency
+
+        self._device_name = self._home_name
+
+    @property
+    def native_value(self):
+        """Return the value of the sensor."""
+        return getattr(self._tibber_home, self.entity_description.key)
+
+    @property
+    def extra_state_attributes(self):
+        """Return the extra state attributes."""
+        if self.entity_description.key == "peak_hour":
+            return {"Time of max hour": self._tibber_home.peak_hour_time}
```

Comment:
Should this be a sensor with device class timestamp?

Reply:
Maybe?
peak_hour is the highest consumption hourly consumption.
peak_hour_time is the time of the highest consumption. (I am not sure it is interesting to do any automation based on this)

So they are strongly connected.
I think it is a. bit unclear when a state_attributes should be used.

Reply:
If a measurement is significant on its own it should be a separate sensor. I'd say "time of the highest consumption" is significant on its own.